hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java

```diff
@@ -116,4 +134,67 @@ public void handleTxBody(@NonNull final HederaState state, @NonNull final Transa
                     e);
         }
     }
+
+    /**
+     * Checks if the throttle definitions are valid.
+     *
+     * @param recordBuilder record builder used to set the status of the transaction
+     */
+    private void throttleValidations(SingleTransactionRecordBuilderImpl recordBuilder) {
```

Comment:
This is the wrong place for such validations. The idea of `SystemFileUpdateFacility` is to check, if a system-file was uploaded, and call the responsible facility. Specific checks should be done in the facility, e.g. `ThrottleManager`. If the validation fails, you have to throw a `HandleException` with the error code. This will ensure that the state changes are rolled back and the `recordBuilder` is updated.